docs/source/dev.rst

```diff
@@ -15,10 +15,14 @@ To release a new version of ``ucc`` on GitHub, follow the steps below.
     - Increment the version in ``pyproject.toml`` according to `semantic versioning <https://semver.org/>`_.
 
 2. **Update the Changelog:**
-    - Update the ``CHANGELOG.rst`` file with all new changes, improvements, and bug fixes since the previous release.
+    - Update the ``CHANGELOG.md`` file with all new changes, improvements, and bug fixes since the previous release.
+    - You can generate an initial set of release notes using GitHub's `draft a release <https://github.com/unitaryfoundation/ucc/releases/new>`_.
+      Note that you are doing this before the actual release tag, since you want the ``CHANGELOG.md`` contents to be updated as of the release tag.
```

Comment:
I tried drafting a a new changelog section per the updated release instructions, but it looks like they only auto-populate if a new tag is created (which we say not to do, for good reason). 

Maybe I'm missing something obvious, but if not I think we should clarify or remove the part about auto-generating release notes.